### DISCUSSION TITLE
cy.reload call not completing even though page was reloaded

### DISCUSSION THREAD

**Discussion by acaen007** [cypress-io/cypress#27271] [Questions and Help]
I am writing a test which needs to reload screen when an element is not present. The behavior is as intended, and reloads when the element is not on the screen, but the reload log never completes and fails the test. This seems to be a cypress issue, was wondering if we could get any support on this?

**Comment by portyu9** [comment]
Hi @acaen007 

Since this is a **Component Test**, `cy.reload()` is the wrong mechanism for resetting the component.

A page reload in Component Testing reloads Cypress's component-test page/spec environment rather than behaving like a normal reload of your production application. Cypress still has an open issue for this exact behavior: reloading during CT can restart the test and get stuck in a loop.

Instead, reset the component through the component-testing API:

- mount the component again with fresh state/dependencies;
- create a new store/service instance for each test instead of importing a singleton;
- use the framework's rerender mechanism when you only need to update props/state.

For example, current React Component Testing exposes a `rerender` function from `mount()`.

If the functionality you actually need to verify is **a real browser-page reload**, that should be an E2E test rather than a Component Test.

So increasing `pageLoadTimeout` or waiting longer is unlikely to address the underlying problem here—the ownership is the Component Testing environment, not the page taking too long to reload.

The full discussion was reviewed; there were no screenshots/attachments or replies.

Please mark this answer as accepted if it helped, thank you.
